2020/09/09/introduction-to-borg-backup.html.md

```diff
@@ -0,0 +1,96 @@
+
+The main objective of Borg is to provide an efficient and secure way to backup data. The deduplication technique utilized to produce the backup process is very quick and effective. 
+
+#### Step 1 - How to install the Borg backups
+On Ubuntu:
+
+`apt install borgbackup`
+
+On Fedora/RHEL:
+
+`dnf install borgbackup`
+
+#### Step 2 - Initialize Local Borg repository
+Firstly, the system that is going to be backed up needs a new designated backup directory. Name the parent directory ‘backup’ and then create a child directory called ‘borgdemo’, which serves as the repository.
+
+`mkdir -p /mnt/backup/`
+`borg init --encryption=repokey /mnt/backup/borgdemo`
+
+#### Step 3 - Let’s create the first backup (archive)
+In Borg terms, each backup instance will be called as an archive. 
+The following demonstrates how to backup the ‘photos’ directory and designate the archive as ‘archive_1’
+
+`borg create —stats —progress /mnt/backup/borgdemo::archive_1 /home/kannan/photos`
+
+**Note: the archive label for each backup run needs to be specified. 
+
+#### Step 4 - Next backup (Incremental)
+In order to see if the run was successful, the same command will be executed again. However, this time, with the different unique archive label.
+
+`borg create —stats —progress /mnt/backup/borgdemo::archive_2 /home/kannan/photos`
+
+The following backup is noticeably identical to the previous one. Because of deduplication, the process will not only run faster this time, it will be incremental as well. 
+
+Executing `—stats`  will provide statistics regarding the size of deduplication
+
+#### Step 5 - List all the archives
+The ‘Borg list’ command lists all of the archives stored within the Borg repository
+
+`borg list /mnt/backup/borgdemo`
+
+#### Step 6 - Remote Borg Repository
+Take the scenario where the backups of many servers need to be maintained in a separate server. In this instance, a directory needs to be created for each of the systems that will be backed up. For this backup repository, create a folder named ‘backup’, and then within ‘Backup,’ a folder called ‘linode_01.’ This folder will be initialized as a Borg repository. 
+
+`/mnt/backup/linode_01` - Server name is ‘linode_01’
+
+`borg init —encryption=repokey user@backup_server:/mnt/backup/linode_01`
+
+Note: the username, backup_server, repo can all be customized at the user’s discretion. 
+
+While initialising the repo, a passphrase for each backup repository can be set for authentication.
+
+#### Step 7 - Create an initial backup to the remote Borg repository
+`borg create --stats ssh://user@backup_server/mnt/backup/linode_01::archive_1 /home/kannan/photos`
+
+To enable the remote backups, the following three environment variables can be used to simplify the automation process:
+- export BORG_REPO=‘ssh://user@backup_server/mnt/backup/linode_01’
+- export BORG_PASSPHRASE=‘set_your_passpharase’
+- export BORG_RSH=’ssh -I /home/kannan/.ssh/id_rsa_backups’
```

Comment:
These quotes are wrong. They should be plain `'` but your editor made them into pretty `‘’` which don't work in the shell.